cwe_checker_rs/src/analysis/backward_interprocedural_fixpoint/tests.rs

```diff
@@ -0,0 +1,245 @@
+    let return_term = Term {
+        tid: Tid::new("return".to_string()),
+        term: Jmp::Return(Expression::Const(Bitvector::zero(64.into()))), // The return term does not matter
+    };
+    let jmp = Jmp::Branch(Tid::new("sub1_blk1"));
+    let jmp_term = Term {
+        tid: Tid::new("jump"),
+        term: jmp,
+    };
+    let sub1_blk1 = Term {
+        tid: Tid::new("sub1_blk1"),
+        term: Blk {
+            defs: vec![def_term1],
+            jmps: vec![call_term],
+        },
+    };
+    let sub1_blk2 = Term {
+        tid: Tid::new("sub1_blk2"),
+        term: Blk {
+            defs: vec![def_term5],
+            jmps: vec![jmp_term],
+        },
+    };
+    let sub1 = Term {
+        tid: Tid::new("sub1"),
+        term: Sub {
+            name: "sub1".to_string(),
+            blocks: vec![sub1_blk1, sub1_blk2],
+        },
+    };
+    let cond_jump = Jmp::CBranch {
+        target: Tid::new("sub1_blk1"),
+        condition: Expression::Const(Bitvector::from_u8(0)),
+    };
+    let cond_jump_term = Term {
+        tid: Tid::new("cond_jump"),
+        term: cond_jump,
+    };
+    let jump_term_2 = Term {
+        tid: Tid::new("jump2"),
+        term: Jmp::Branch(Tid::new("sub2_blk2")),
+    };
+    let sub2_blk1 = Term {
+        tid: Tid::new("sub2_blk1"),
+        term: Blk {
+            defs: vec![def_term2, def_term3],
+            jmps: vec![cond_jump_term, jump_term_2],
+        },
+    };
+    let sub2_blk2 = Term {
+        tid: Tid::new("sub2_blk2"),
+        term: Blk {
+            defs: vec![def_term4],
+            jmps: vec![return_term],
+        },
+    };
+    let sub2 = Term {
+        tid: Tid::new("sub2"),
+        term: Sub {
+            name: "sub2".to_string(),
+            blocks: vec![sub2_blk1, sub2_blk2],
+        },
+    };
+    let program = Term {
+        tid: Tid::new("program"),
+        term: Program {
+            subs: vec![sub1, sub2],
+            extern_symbols: Vec::new(),
+            entry_points: Vec::new(),
+        },
+    };
+    program
+}
+
+#[test]
+fn backward_fixpoint() {
+    let project = Project {
+        program: mock_program(),
+        cpu_architecture: String::from("x86"),
+        stack_pointer_register: Variable {
+            name: String::from("RSP"),
+            size: ByteSize::new(8),
+            is_temp: false,
+        },
+        calling_conventions: Vec::new(),
+    };
+
+    let mock_con = Context::new(&project);
+    let mut computation = Computation::new(mock_con, None);
+    computation.set_node_value(NodeIndex::new(0), NodeValue::Value(0));
+    computation.compute_with_max_steps(100);
+
+    // The fixpoint values of all 12 BlockStart/BlockEnd nodes are compared with their expected value
+    assert_eq!(
+        *computation
+            .get_node_value(NodeIndex::new(0))
+            .unwrap()
+            .unwrap_value(),
+        0 as u64
+    );
+    assert_eq!(
+        *computation
+            .get_node_value(NodeIndex::new(1))
```

Comment:
Hmm... This could fail if some unrelated change in `graph.rs` also changes the order in which nodes are added to the graph. Why not identify them by the block TIDs instead?